pkg/apis/components/v1alpha1/types.go

```diff
@@ -34,6 +34,7 @@ type ComponentSpec struct {
 	Type string `json:"type"`
 	// +optional
 	Version  string         `json:"version"`
+	Critical bool           `json:"critical"`
```

Comment:
Add kubebuilder default value here, so it defaults to true. Another way, would be to rename it to NonCritical and leave the default as false.

Reply:
I wanted to avoid double negation like `noncritical: false`. 
Should I create a property _trivial_ and set its default value to false?

Reply:
What about `Optional`? It can default to false.

Reply:
I would use a more descriptive name: `failOnError`